webrtc: remove setRemoteDescription / msid test

fippo opened the pull request:
since track ids are essentially meaningless. If the party calling
setRemoteDescription would have added a track before calling SRD
the assertion from this test would break.

See https://github.com/w3c/webrtc-pc/issues/1128 for some discussion.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/5602)
<!-- Reviewable:end -->

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223298137)

# Lint #

Passed

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223298139)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 5ee726463376166c18eb25c4eb55ef4cced3fd1c
Using browser at version BuildID 20170417100236; SourceStamp c697e756f738ce37abc56f31bfbc48f55625d617
Starting 10 test iterations
No tests run.

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223298140)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 5ee726463376166c18eb25c4eb55ef4cced3fd1c
Using browser at version 59.0.3067.0 dev
Starting 10 test iterations
No tests run.

foolip commented:
The existing test does pass on Firefox, is there nothing of value that is being asserted? The spec presumably requires *some* behavior, even if the test is currently wrong?

agouaillard-cosmo commented:
I cannot see the comments from, get notifications about or send messages to @fippo. It makes reviewing more troublesome than it should. @fippo, could you make the necessary changes so we can proceed with this PR and all. your future contributions?

foolip commented:
@agouaillard-cosmo, what do you mean? @fippo hasn't written any comments on this PR.

This PR has conflicts though, because I moved the file in question. Sorry?

agouaillard commented:
I did not get the notification of the PR creation, and of the first message, and I cannot exchange with him (using @fippo). I suppose he is blocking me.

fippo commented:
@foolip: it passed in Chrome with adapter too. In general, the msid to track mapping is broken since adding transceivers. While the particular case tested would probably continue to work it doesn't add value anymore. Rebased.

w3c-bots commented:
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5602/)

foolip commented:
OK, so this is beyond my knowledge of WebRTC. @henbos, can you check the test and confirm that it's not testing anything valuable?

henbos commented:
> If the party calling setRemoteDescription would have added a track before calling SRD the assertion from this test would break.

But the test does not add a track before calling setRemoteDescription, we know that. It tests a very basic case of one SRD call using msid. It looks like it is testing https://tools.ietf.org/html/draft-ietf-mmusic-msid-16#section-3. Am I missing something? (Not an SDP expert.)

agouaillard commented:
This is not adapter.js, we are not here to address implementation problems. We are here to test against the specs. The fact that browser implementation has been recently broken does not matter. W3C tests can break and should break when the browsers are not compliant.

Does the code test an existing API or feature against its normative definition? Unless it does not, the test should NOT be removed.

foolip commented:
@fippo, in https://github.com/w3c/webrtc-pc/issues/1128#issuecomment-294963117 you wrote:
> However, it does not hold if the receiver called addTrack before calling setRemoteDescription which causes a RTPReceiver to be creating with a random track id which is then used in ontrack. I'm going to remove the wpt test which made that assertion.

However, this test doesn't call addTrack. If the test is asserting something about MSID that isn't guaranteed, can you remove just that part?

fippo committed:
webrtc: remove track parsing assertion from setRemoteDescription/msid test

removes the assertion that the track id in ontrack comes from
the SDP. While that is still true for this test, it is not for others.

See https://github.com/w3c/webrtc-pc/issues/1128

fippo commented:
@foolip done.

agouaillard commented:
LGTM.
